common/sgx/tlsverifier.c

```diff
@@ -172,3 +197,121 @@ oe_result_t oe_verify_attestation_certificate(
     oe_free(report);
     return result;
 }
+
+/**
+ * oe_verify_plugin_attestation_certificate
+ *
+ * This function perform a custom validation on the input certificate. This
+ * validation includes extracting an attestation evidence extension from the
+ * certificate before validating this evidence. An optional
+ * claim_verify_callback could be passed in for a calling client to further
+ * validate the claims of the enclave creating the certificate.
+ * @param[in] cert_in_der a pointer to buffer holding certificate contents
+ *  in DER format
+ * @param[in] cert_in_der_len size of certificate buffer above
+ * @param[in] claim_verify_callback callback routine for custom claim checking
+ * @param[in] arg an optional context pointer argument specified by the caller
+ * when setting callback
+ * @retval OE_OK on a successful validation
+ * @retval OE_VERIFY_FAILED on quote failure
+ * @retval OE_INVALID_PARAMETER At least one parameter is invalid
+ * @retval OE_FAILURE general failure
+ * @retval other appropriate error code
+ */
+oe_result_t oe_verify_plugin_attestation_certificate(
```

Comment:
Suggest 'oe_verify_attestation_evidence_certificate'. Same in all occurrences.

Reason:
1. Embedded attestation evidence is generated by oe_get_evidence().

Reply:
I think the goal of this API is to allow different plugins to generate their own certificates. Evidence is just the means to achieve a generic representation of proof provided by different plugins.

Reply:
As this will become a new public API to OE, I am open to suggestions of better names.

Reply:
I don't understand this PR enough to comment, is there a design doc or diagram?   E.g., is the certificate part of the evidence?   (It certainly is in CyRes attestation.)   If its a public API without _sgx_ in the name, then it should be applicable to other TEEs too, and I can't tell yet whether this PR only makes sense for the SGX plugin.

Reply:
I agree with @yentsanglee's reasoning and left a similar comment in the header files before I saw this one; either suggested name seems fine to me, though Yen Lee's is shorter. I did make some other related suggestions there as well (e.g. `get` vs. `generate`)

Reply:
Renamed to `oe_get_attestation_certificate_with_evidence` as @CodeMonkeyLeet suggested. Also moved the declaration to attester.h